ultralytics/data/annotator.py

```diff
@@ -64,3 +71,337 @@ def auto_annotate(
+
+
+class Florence2:
+    """
+    A Florence-2 model wrapper for grounding-style object detection and auto-annotation.
+
+    This class handles loading Florence2 variants and running image-text inference to produce object bounding boxes
+    and labels. It integrates with Ultralytics' AutoAnnotator for automatic dataset labeling.
+
+    Attributes:
+        device (torch.device): Inference device (e.g., 'cuda', 'cpu').
+        variant (str): Florence2 variant used (e.g., "base", "large-ft").
+        mid (str): Model identifier string used with Hugging Face.
+        model (torch.nn.Module): Loaded Florence2 model.
+        processor (Florence2Processor): Image and text processor used for inference.
+
+    Methods:
+        _load_model: Internal method to load Florence2 model and processor.
+        process: Performs inference and returns bounding boxes and labels in YOLO-compatible format.
+    """
+
+    def __init__(self, device="", variant="base"):
+        """
+        Initializes the Florence2 model with the specified variant and device.
+
+        Args:
+            device (str): Target device for inference, e.g., 'cuda', 'cpu'.
+            variant (str): Model variant to use: "base", "large", "base-ft", "large-ft".
+        """
+        self.device = select_device(device)
+
+        supported_variants = {"base", "large", "large-ft", "base-ft"}
+        self.variant = variant if variant in supported_variants else "base"
+
+        if variant not in supported_variants:
+            LOGGER.warning(
+                f"⚠️ Invalid variant '{variant}' provided. Falling back to 'base'. "
+                f"Supported variants: {supported_variants}"
+            )
+
+        self.mid = f"microsoft/florence-2-{self.variant}"
+        self.model = None
+        self.processor = None
+        self._load_model()
+
+    def _load_model(self):
+        """
+        Loads the Florence2 model and its corresponding processor from Hugging Face Hub.
+
+        This method requires `transformers` and `einops`. It sets up the model and processor
+        using `trust_remote_code=True`, which allows Florence2 custom architecture loading.
+
+        Returns:
+            None
+        """
+        from ultralytics.utils.checks import check_requirements
+
+        check_requirements(["transformers==4.49.0", "einops"])
+
+        from transformers import AutoModelForCausalLM, AutoProcessor, logging
+
+        logging.set_verbosity_error()  # Minimize Hugging Face logs
+
+        LOGGER.info(f"💡 Initializing Florence2-{self.variant} on {str(self.device).upper()}")
+
+        self.model = AutoModelForCausalLM.from_pretrained(
+            self.mid,
+            trust_remote_code=True,
+        ).to(self.device)
+
+        self.processor = AutoProcessor.from_pretrained(
+            self.mid,
+            trust_remote_code=True,
+        )
+
+        LOGGER.info(f"🚀 Loaded Florence2-{self.variant} model successfully.")
```

Comment:
Remove emoji. Model loading never has an emoji in Ultralytics. Consistency please.